src/lib/containers/EntityLink.tsx

```diff
@@ -1,31 +1,66 @@
+import { Skeleton } from '@material-ui/lab'
 import React from 'react'
-import { getEntityTypeFromHeader } from '../utils/functions/EntityTypeUtils'
+import {
+  convertToEntityType,
+  getEntityTypeFromHeader,
+} from '../utils/functions/EntityTypeUtils'
 import { PRODUCTION_ENDPOINT_CONFIG } from '../utils/functions/getEndpoint'
-import { EntityHeader } from '../utils/synapseTypes/'
+import { useGetEntity } from '../utils/hooks/SynapseAPI/useEntity'
+import { Entity, EntityHeader } from '../utils/synapseTypes/'
 import { EntityTypeIcon } from './EntityIcon'
 
 type EntityLinkProps = {
-  entityHeader: EntityHeader
+  entity: string | EntityHeader | Entity
+  link?: boolean
   className?: string
 }
 
-export const EntityLink: React.FC<EntityLinkProps> = ({
-  entityHeader,
-  className,
-}) => {
-  const { id, name } = entityHeader
-  const type = getEntityTypeFromHeader(entityHeader)
-  return (
-    <p className={className}>
-      <a
-        className={className}
-        target="_blank"
-        rel="noopener noreferrer"
-        href={`${PRODUCTION_ENDPOINT_CONFIG.PORTAL}#!Synapse:${id}`}
-      >
-        <EntityTypeIcon type={type} style={{ marginRight: '5px' }} />
-        {name}
-      </a>
-    </p>
-  )
+export const EntityLink = (props: EntityLinkProps) => {
+  const { entity: entityOrId, className, link = true } = props
+
+  let entityId = ''
+  if (typeof entityOrId === 'string') {
+    entityId = entityOrId
+  }
+
+  const { data: fetchedEntity } = useGetEntity(entityId, undefined, {
+    enabled: !!entityId && typeof entityOrId === 'string',
+  })
+
+  if (fetchedEntity || typeof entityOrId !== 'string') {
+    const entity = fetchedEntity ?? (entityOrId as Entity | EntityHeader)
+    let type
+    if ('concreteType' in entity) {
+      type = convertToEntityType(entity.concreteType)
+    } else {
+      type = getEntityTypeFromHeader(entity)
+    }
+
+    if (link) {
+      return (
+        <p className={className}>
+          <a
+            className={className}
+            target="_blank"
+            rel="noopener noreferrer"
+            href={`${PRODUCTION_ENDPOINT_CONFIG.PORTAL}#!Synapse:${entity.id!}`}
+          >
+            <EntityTypeIcon type={type} style={{ marginRight: '5px' }} />
+            {entity.name}
+          </a>
+        </p>
+      )
+    } else {
+      return (
+        <p className={className}>
+          <EntityTypeIcon type={type} style={{ marginRight: '5px' }} />
+          {entity.name}
+        </p>
+      )
+    }
+  } else {
+    // Entity has not been fetched yet
+    // TODO: Handle not found, unauthorized
```

Comment:
Do we have any existing components/patterns for these cases?

Reply:
As you know, we have an ErrorBanner, which is not a good fit for a table cell entity load error.  I'd recommend the UI proposed for files that are ineligible for package download, found [here](https://www.figma.com/file/usEyd7hdEd0uYyN04yt0WO/branch/PnQZRd4YeLhvFj6IVSaVQb/Data-Lists-for-Download-R1?node-id=2282%3A443865).  Where the error would be displayed in the tooltip.
<img width="644" alt="Screen Shot 2021-11-18 at 10 25 12 AM" src="https://user-images.githubusercontent.com/1864447/142474635-7cd85107-5276-473a-882d-2a3d963e2338.png">